python/ray/experimental/data/dataset.py

```diff
@@ -1111,21 +1127,34 @@ def to_arrow(self) -> List[ObjectRef["pyarrow.Table"]]:
         """Convert this dataset into a distributed set of Arrow tables.
 
         This is only supported for datasets convertible to Arrow records.
-        This function induces a copy of the data. For zero-copy access to the
-        underlying data, consider using ``.get_blocks()`` instead.
+        This function is zero-copy if the existing data is already in Arrow
+        format. Otherwise, the data will be converted to Arrow format.
 
-        Time complexity: O(dataset size / parallelism)
+        Time complexity: O(1) unless conversion is required.
 
         Returns:
             A list of remote Arrow tables created from this dataset.
         """
 
         @ray.remote
-        def block_to_df(block: ArrowBlock):
+        def check_is_arrow(block: Block) -> bool:
+            import pyarrow
+            return isinstance(block, pyarrow.Table)
+
+        blocks: List[ObjectRef[Block[T]]] = list(self._blocks)
+        is_arrow = ray.get(check_is_arrow.remote(blocks[0]))
```

Comment:
Why not block directly on the block ref?

```python
import pyarrow as pa

block = next(iter(self._blocks))
if isinstance(ray.get(block), pa.Table):
    return list(self._blocks)
```

Reply:
We don't want to fetch the block locally here.

Reply:
Ah, that makes sense.